L1Trigger/DTTriggerPhase2/interface/MPCleanHitsFilter.h

```diff
@@ -0,0 +1,51 @@
+#ifndef L1Trigger_DTTriggerPhase2_MPCleanHitsFilter_h
+#define L1Trigger_DTTriggerPhase2_MPCleanHitsFilter_h
+
+#include "L1Trigger/DTTriggerPhase2/interface/MPFilter.h"
+
+#include <iostream>
+#include <fstream>
+
+// ===============================================================================
+// Previous definitions and declarations
+// ===============================================================================
+
+// ===============================================================================
+// Class declarations
+// ===============================================================================
+
+class MPCleanHitsFilter : public MPFilter {
+public:
+  // Constructors and destructor
+  MPCleanHitsFilter(const edm::ParameterSet& pset);
+  ~MPCleanHitsFilter() override;
+
+  // Main methods
+  void initialise(const edm::EventSetup& iEventSetup) override{};
+  void run(edm::Event& iEvent,
+           const edm::EventSetup& iEventSetup,
+           std::vector<cmsdt::metaPrimitive>& inMPath,
+           std::vector<cmsdt::metaPrimitive>& outMPath) override{};
+
+  void run(edm::Event& iEvent,
+           const edm::EventSetup& iEventSetup,
+           MuonPathPtrs& inMPath,
+           MuonPathPtrs& outMPath) override;
+
+  void finish() override{};
+
+  // Other public methods
+  void removeOutliers(MuonPathPtr& mpath);
+
+  double getMeanTime(MuonPathPtr& mpath);
+
+  void setTimeTolerance(int time) { timeTolerance_ = time; }
+  int getTimeTolerance() { return timeTolerance_; }
+
+private:
+  // Private attributes
+  bool debug_;
```

Comment:
```suggestion
  const bool debug_;
```

Reply:
> Sorry if I make an stupid question, but can this be done even if debug_ gets modified in the .cc? (i.e. https://github.com/dtp2-tpg-am/cmssw/blob/AM_12_3_X_integration/L1Trigger/DTTriggerPhase2/src/MPQualityEnhancerFilterBayes.cc#L14)

Good point!
You can initialize it following what I scratched now in https://github.com/cms-sw/cmssw/pull/37237/files#r857679108